iOSNativeLoginTemplate/iOSNativeLoginTemplate/SceneDelegate.swift

```diff
@@ -0,0 +1,123 @@
+WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import UIKit
+import SwiftUI
+import MobileSync
+import SalesforceSDKCore
+
+class SceneDelegate: UIResponder, UIWindowSceneDelegate {
+
+    var window: UIWindow?
+    
+    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+        
+        guard let windowScene = (scene as? UIWindowScene) else { return }
+        self.window = UIWindow(frame: windowScene.coordinateSpace.bounds)
+        self.window?.windowScene = windowScene
+        
+        // This Client Id and Redirect Url is specific and exclusive to Native Login.
+        let clientId = "3MVG9CEn_O3jvv0wTqRT0Le6tmzX.EQ9ZvtHL1TG3gHFV.4IvKZyXw5SgdiVPi61mXrpu40mCOhKevEfYNMOm"
+        let redirectUri = "https://msdk-enhanced-dev-ed.my.site.com/services/oauth2/echo"
+        let loginUrl = "https://msdk-enhanced-dev-ed.my.site.com/headless"
+        // Used to create a View Controller from SwiftUI.
+        let nativeLoginViewConroller = NativeLoginViewFactory.create()
+
+        // This line tells the SDK the app intends to use Native Login.
+        SalesforceManager.shared.useNativeLogin(withConsumerKey: clientId, callbackUrl: redirectUri, communityUrl: loginUrl,
+                                                nativeLoginViewController: nativeLoginViewConroller, scene:scene)
+       
+        AuthHelper.registerBlock(forCurrentUserChangeNotifications: {
+           self.resetViewState {
+               self.setupRootViewController()
+           }
+        })
+    }
+
+    func sceneDidDisconnect(_ scene: UIScene) {
+        // Called as the scene is being released by the system.
+        // This occurs shortly after the scene enters the background, or when its session is discarded.
+        // Release any resources associated with this scene that can be re-created the next time the scene connects.
+        // The scene may re-connect later, as its session was not neccessarily discarded (see `application:didDiscardSceneSessions` instead).
+    }
+
+    func sceneDidBecomeActive(_ scene: UIScene) {
+        // Called when the scene has moved from an inactive state to an active state.
+        // Use this method to restart any tasks that were paused (or not yet started) when the scene was inactive.
+    }
+
+    func sceneWillResignActive(_ scene: UIScene) {
+        // Called when the scene will move from an active state to an inactive state.
+        // This may occur due to temporary interruptions (ex. an incoming phone call).
+    }
+
+    func sceneWillEnterForeground(_ scene: UIScene) {
+        // Called as the scene transitions from the background to the foreground.
+        // Use this method to undo the changes made on entering the background.
+        self.initializeAppViewState()
+        AuthHelper.loginIfRequired {
```

Comment:
how does this work with the native login flow?

Reply:
Yes, like webview login _it is required_.  Native Login is launched through the same `loginWithCompletion` codepath as the webview.